sdk/monitor/azure-monitor-ingestion/azure/monitor/ingestion/_helpers.py

```diff
@@ -2,40 +2,41 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 # ------------------------------------
-
 import json
+import logging
+from typing import Generator, List
 import zlib
 
-MAX_CHUNK_SIZE = 1000000
-CHAR_SIZE = 4
+_LOGGER = logging.getLogger(__name__)
+
+MAX_CHUNK_SIZE_BYTES = 1024 * 1024 # 1 MiB
+CHAR_SIZE_BYTES = 4
 
 
-def _split_chunks(logs):
-    chunks = []
+def _split_chunks(logs: List) -> Generator:
     chunk_size = 0
     curr_chunk = []
     for log in logs:
         # each char is 4 bytes
-        size = len(json.dumps(log)) * CHAR_SIZE
-        if chunk_size + size < MAX_CHUNK_SIZE:
+        size = len(json.dumps(log)) * CHAR_SIZE_BYTES
+        if chunk_size + size < MAX_CHUNK_SIZE_BYTES:
             curr_chunk.append(log)
             chunk_size += size
         else:
-            chunks.append(curr_chunk)
+            _LOGGER.debug('Yielding chunk with size: %d', chunk_size)
+            yield curr_chunk
             curr_chunk = [log]
             chunk_size = size
     if len(curr_chunk) > 0:
-        chunks.append(curr_chunk)
-    return chunks
+        _LOGGER.debug('Yielding chunk with size: %d', chunk_size)
+        yield curr_chunk
 
 
-def _create_gzip_requests(logs):
-    requests = []
-    chunks = _split_chunks(logs)
-    for chunk in chunks:
+def _create_gzip_requests(logs: List) -> Generator:
```

Comment:
The type hint in the public `upload` method states that we accept `IO` as an input type for logs?
This would probably work if the IO data is structured with one log entry per line because I think a file IO will iterate lines by default...
Do we have tests for this? Does it work with any unstructured IO data passed in?

Reply:
I think the `IO` type was prematurely added during the transition just before I took over, and I don't think it was tested. Next on my list is ensure file streams can be handled, so I will remove the `IO` typing for now and then re-add it once I've verified and added io tests for upload in a subsequent PR. Hopefully that works.